openapi-cli/src/main/java/io/ballerina/openapi/generators/client/BallerinaClientGenerator.java

```diff
@@ -187,8 +195,15 @@ public SyntaxTree generateSyntaxTree() throws BallerinaOpenApiException {
             // Add authentication related records to module member nodes
             nodes.add(ballerinaAuthConfigGenerator.getConfigRecord(openAPI));
         }
+
+        // Add `AuthConfig` record if combination of ApiKey and HTTP/OAuth authentication is supported
+        if (ballerinaAuthConfigGenerator.isApiKey() && ballerinaAuthConfigGenerator.isHttpOROAuth()) {
+            nodes.add(ballerinaAuthConfigGenerator.getAuthConfigRecord());
```

Comment:
Shall we add a function at `BallerinaAuthConfigGenerator` to return all the authentication related records at once ? Will be more readable in that way. WDYT ?